journal/src/main/java/io/camunda/zeebe/journal/file/SegmentedJournal.java

```diff
@@ -493,57 +496,110 @@ private FileChannel openChannel(final File file) {
   protected Collection<JournalSegment> loadSegments() {
     // Ensure log directories are created.
     directory.mkdirs();
+    final List<JournalSegment> segments = new ArrayList<>();
 
-    final TreeMap<Long, JournalSegment> segments = new TreeMap<>();
+    final List<File> files = getSortedLogSegments();
+    for (int i = 0; i < files.size(); i++) {
+      final File file = files.get(i);
 
-    // Iterate through all files in the log directory.
-    for (final File file : directory.listFiles(File::isFile)) {
+      try {
+        final JournalSegmentDescriptor descriptor = readDescriptor(file);
+        final JournalSegment segment = loadSegment(descriptor.id());
 
-      // If the file looks like a segment file, attempt to load the segment.
-      if (JournalSegmentFile.isSegmentFile(name, file)) {
-        final JournalSegmentFile segmentFile = new JournalSegmentFile(file);
-        final ByteBuffer buffer = ByteBuffer.allocate(JournalSegmentDescriptor.getEncodingLength());
-        try (final FileChannel channel = openChannel(file)) {
-          channel.read(buffer);
-          buffer.flip();
-        } catch (final IOException e) {
-          throw new JournalException(e);
+        if (i > 0 && segments.get(i - 1).lastIndex() + 1 != segment.index()) {
+          throw new CorruptedLogException(
+              String.format(
+                  "Log segment %s is not aligned with previous segment %s.",
+                  segments.get(i - 1), segment));
         }
 
-        final JournalSegmentDescriptor descriptor = new JournalSegmentDescriptor(buffer);
-
-        // Load the segment.
-        final JournalSegment segment = loadSegment(descriptor.id());
+        log.debug("Found segment: {} ({})", descriptor.id(), file.getName());
```

Comment:
May be we can move this log before reading the descriptor. This way if some thing fails when reading the segment, we know which segmented failed. Currently, both log messages - "Loaded segment" and "Found segment" are redundant and printed after the segment is successfully loaded.